contracts/lens/Quoter.sol

```diff
@@ -0,0 +1,270 @@
+                mstore(add(ptr, 0x20), sqrtPriceX96After)
+                mstore(add(ptr, 0x40), tickAfter)
+                revert(ptr, 96)
+            }
+        } else if (swapInfo.swapType == SwapType.ExactInput) {
+            (
+                int128[] memory deltaAmounts,
+                uint160[] memory sqrtPriceX96AfterList,
+                uint32[] memory initializedTicksCrossedList
+            ) = _quoteExactInput(abi.decode(swapInfo.params, (ExactInputParams)));
+
+            assembly {
+                // function storeArray(offset, length, array) {
+                //     mstore(offset, length)
+                //     offset := add(offset, 0x20)
+                //     for { let i := 0 } lt(i, length) { i := add(i, 1) } {
+                //         let value := mload(add(array, add(mul(i, 0x20), 0x20)))
+                //         mstore(offset, value)
+                //         offset := add(offset, 0x20)
+                //     }
+                // }
+
+                let originalPtr := mload(0x40)
+                let ptr := mload(0x40)
+
+                let deltaLength := mload(deltaAmounts)
+                let sqrtPriceLength := mload(sqrtPriceX96AfterList)
+                let initializedTicksLength := mload(initializedTicksCrossedList)
+
+                let deltaOffset := 0x60
+                let sqrtPriceOffset := add(deltaOffset, add(0x20, mul(0x20, deltaLength)))
+                let initializedTicksOffset := add(sqrtPriceOffset, add(0x20, mul(0x20, sqrtPriceLength)))
+
+                // storing offsets to dynamic arrays
+                mstore(ptr, deltaOffset)
+                ptr := add(ptr, 0x20)
+                mstore(ptr, sqrtPriceOffset)
+                ptr := add(ptr, 0x20)
+                mstore(ptr, initializedTicksOffset)
+                ptr := add(ptr, 0x20)
+
+                //storing length + contents of dynamic arrays
+                mstore(ptr, deltaLength)
+                ptr := add(ptr, 0x20)
+                for { let i := 0 } lt(i, deltaLength) { i := add(i, 1) } {
+                    let value := mload(add(deltaAmounts, add(mul(i, 0x20), 0x20)))
+                    mstore(ptr, value)
+                    ptr := add(ptr, 0x20)
+                }
+
+                mstore(ptr, sqrtPriceLength)
+                ptr := add(ptr, 0x20)
+                for { let i := 0 } lt(i, sqrtPriceLength) { i := add(i, 1) } {
+                    let value := mload(add(sqrtPriceX96AfterList, add(mul(i, 0x20), 0x20)))
+                    mstore(ptr, value)
+                    ptr := add(ptr, 0x20)
+                }
+
+                mstore(ptr, initializedTicksLength)
+                ptr := add(ptr, 0x20)
+                for { let i := 0 } lt(i, initializedTicksLength) { i := add(i, 1) } {
+                    let value := mload(add(initializedTicksCrossedList, add(mul(i, 0x20), 0x20)))
+                    mstore(ptr, value)
+                    ptr := add(ptr, 0x20)
+                }
```

Comment:
sheeesshh nice assembly ahah. Though I'm a bit surprised you can't do something more like

```
bytes memory result = abi.encode(deltaAmounts, sqrtPrice, initializedTicksLength);
assembly {
   revert(result, result.length)
}
```
to get the same result